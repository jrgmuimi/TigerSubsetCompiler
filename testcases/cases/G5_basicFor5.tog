let
    var loop_ctrl : int
    var loop_ctrl1 : int
    var loop_ctrl2 : int
in
    for loop_ctrl:=0 to 10 do 
        printint(loop_ctrl);
        printstring("loop1\n")
    end;

    for loop_ctrl1:=0 to loop_ctrl do 
        printint(loop_ctrl1);
        printstring("loop2\n")
    end;


    for loop_ctrl2:=0 to loop_ctrl1 do 
        printint(loop_ctrl2);
        printstring("loop3\n")
    end;

    for loop_ctrl:=0 to loop_ctrl2 do 
        printint(loop_ctrl);
        printstring("loop4\n")
    end

end