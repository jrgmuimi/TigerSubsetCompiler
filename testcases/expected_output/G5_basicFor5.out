MARS 4.5  Copyright 2003-2014 Pete Sanderson and Kenneth Vollmar

0loop1
1loop1
2loop1
3loop1
4loop1
5loop1
6loop1
7loop1
8loop1
9loop1
10loop1
0loop2
1loop2
2loop2
3loop2
4loop2
5loop2
6loop2
7loop2
8loop2
9loop2
10loop2
11loop2
0loop3
1loop3
2loop3
3loop3
4loop3
5loop3
6loop3
7loop3
8loop3
9loop3
10loop3
11loop3
12loop3
0loop4
1loop4
2loop4
3loop4
4loop4
5loop4
6loop4
7loop4
8loop4
9loop4
10loop4
11loop4
12loop4
13loop4

